### PR TITLE
Custom filter on specific properties using ApiFilter annotation

### PR DESCRIPTION
**Pull request by vvatelot**
When using `ApiFilter` annotation, it seems that you have to set the concerned properties in the annotation itself.

### CODE REVIEW

**Comment by soyuka** [comment]
Isn't this explained in the first section of https://api-platform.com/docs/core/filters/#apifilter-annotation ?

**Comment by vvatelot** [comment]
@soyuka yes you're right!
My point is, when reading the documentation (to create a custom filter), I follow the steps:

1. Creating the filter class
2. Declare the filter in the `services.yaml`
3. Chose between `ApiResource` or `ApiFilter` annotation

In my case, I prefer to use `ApiFilter`, but in this case the properties declared in the arguments of `services.yaml` are not taken into account. So yes, after taking some time to find it out it is now clear that when using `ApiFilter` you have to refer to the https://api-platform.com/docs/core/filters/#apifilter-annotation section.

So, my proposal is maybe not accurate, and it should just mention what I mention here with a referal to the https://api-platform.com/docs/core/filters/#apifilter-annotation section

What do you think?

**Comment by soyuka** [comment]
Yeah totally we could add a reference in the custom filters doc

File: core/filters.md [hunk 1]
```diff
@@ -950,8 +950,9 @@ class Offer
     // ...
 }
 ```
+When using `ApiFilter` annotation, the declared properties in the `services.yaml` will not be taken into account. You have to use the `ApiFilter` way (see the [documentation](#apifilter-annotation))
```
Comment (alanpoulain) [inline core/filters.md]:
```suggestion
When using `ApiFilter` annotation, the declared properties in the `services.yaml` will not be taken into account. You have to use the `ApiFilter` way (see the [documentation](#apifilter-annotation)).
```

**Comment by soyuka** [comment]
Thanks @vvatelot @alanpoulain !

**Pull request merged**